pkg/storage/bucket/client.go

```diff
@@ -39,17 +41,21 @@ const (
 
 	// Filesystem is the value for the filesystem storage backend.
 	Filesystem = "filesystem"
+
+	validPrefixCharactersRegex = `^[\da-zA-Z\-_.*'()!]+$`
```

Comment:
I am surprised that we allow characters like: `*'()!` here. On the other hand, why do we even check for all of these characters? We really only care about `/` because we use that as delimiter. Other characters should be fine, as long as bucket supports them?

Reply:
But yeah, I really think we shouldn't validate character sets. We don't know what works with different storage providers and we don't know if customers have some specific needs. I don't have experience with fighting the subtle bugs that @andyasp [mentioned](https://github.com/grafana/mimir/pull/1686#discussion_r850461656)

Reply:
Also strictly following the safe character set isn't necessary. For instance it could just be alphanumeric if that's easier to express. Again, this is all just a suggestion in case it's becoming frustrating to handle.

Reply:
I prefer the idea of going with alphanumeric-only. It's easier to explain and validate

Reply:
added in d3c95f805aaf3967e63f310398f47ac55abdecf6